pybossa/cache/task_browse_helpers.py

```diff
@@ -302,6 +292,30 @@ def parse_tasks_browse_args(args):
 
     return parsed_args
 
+def parse_tasks_browse_order_by_args(order_by, display_info_columns):
+    order_by_result = None
+    order_by_dict = dict()
+
+    if order_by:
+        order_by_result = order_by.strip()
+
+        # allowing custom user added task.info columns to be sortable
+        allowed_sort_fields = allowed_fields.copy()
+        allowed_sort_fields.update({col: "task.info->>'{}'".format(col) for col in display_info_columns})
+        for clause in order_by.split(','):
+            order_by_field = clause.split(' ')
+            if len(order_by_field) != 2 or order_by_field[0] not in allowed_sort_fields:
+                raise ValueError('order_by value sent by the user is invalid: %s'.format(order_by))
+            if order_by_field[0] in order_by_dict:
+                raise ValueError('order_by field is duplicated: %s'.format(order_by))
+            order_by_dict[order_by_field[0]] = order_by_field[1]
+
+        for key, value in allowed_sort_fields.items():
```

Comment:
The loop of the allowed_sort_fields dictionary here seems unnecessary. How about changing it to the following:
```
item, order = order_by.split()
order_by_result = f"{allowed_sort_fields[item]} {order}"
```

Reply:
I may not be as familiar with this syntax. However, this code was pre-existing and moved inside the new function for unit testing.

Reply:
Here I'm using f-sting, basically it's the same as xxxx.format(). The idea is to get rid of the unnecessary loop over dictionary. Instead, just get the value the dictionary.
e.g.
```
>>> allowed_sort_fields = {'bi': "task.info->>'bi'", 'companyBbid': "task.info->>'
>>> order_by = 'companyBbid desc'
>>> order_by.split()
['companyBbid', 'desc']
>>> item, order = order_by.split()
>>> item
'companyBbid'
>>> order
'desc'
>>> order_by_result = f"{allowed_sort_fields[item]} {order}"
>>> order_by
'companyBbid desc'
```

But to support multiple columns, the code would need to be placed under ```for clause in order_by.split(','):```

Reply:
Fixed via https://github.com/bloomberg/pybossa/pull/719/commits/6e414a566457c427b1cc4fe83fc275687eacc352